File: LifeOS/Sources/Core/Models/TransactionItem.swift
import Foundation
import SwiftData

@Model
final class TransactionItem {
    var title: String
    var amount: Double
    var date: Date
    var category: String
    var isExpense: Bool
    
    init(title: String, amount: Double, date: Date = Date(), category: String = "Other", isExpense: Bool = true) {
        self.title = title
        self.amount = amount
        self.date = date
        self.category = category
        self.isExpense = isExpense
    }
}

File: android/app/src/main/java/com/yashlunawat/lifeos/data/models/TransactionItem.kt
package com.yashlunawat.lifeos.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
data class TransactionItem(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val amount: Double,
    val date: Long = System.currentTimeMillis(),
    val category: String = "Other",
    val isExpense: Boolean = true,
    val isRecurring: Boolean = false,
    val iconName: String = "creditcard"
) {
    val categoryEmoji: String
        get() = when (category.lowercase()) {
            "food", "dining" -> "🍕"
            "transport", "travel" -> "🚗"
            "shopping" -> "🛍️"
            "entertainment" -> "🎮"
            "bills", "utilities" -> "💡"
            "health" -> "💊"
            "education" -> "📚"
            "subscription" -> "🔄"
            "income", "salary" -> "💰"
            else -> "💳"
        }
}
